File: carla_ros2_client/carla_client.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import rclpy                                   # ROS2 Python接口库
from rclpy.node import Node                    # ROS2 节点类
import time

def main(args=None):                           # ROS2节点主入口main函数
    rclpy.init(args=args)                      # ROS2 Python接口初始化
    node = Node("carla_client")             # 创建ROS2节点对象并进行初始化

    while rclpy.ok():                          # ROS2系统是否正常运行
        node.get_logger().info('clarencez: carla client')  # ROS2日志输出
        time.sleep(0.5)                        # 休眠控制循环时间

    node.destroy_node()                        # 销毁节点对象
    rclpy.shutdown()                           # 关闭ROS2 Python接口

if __name__ == '__main__':
    main()

File: carla_client_cpp/src/carla_client.cpp
#include "rclcpp/rclcpp.hpp"
#include <unistd.h>

int main(int argc, char **argv)
{
    //init client
    rclcpp::init(argc, argv);
    //create new node
    auto node = std::make_shared<rclcpp::Node>("carla_cpp_client");

    while(rclcpp::ok())
    {
        RCLCPP_INFO(node->get_logger(), "clarencez: carla client cpp run");
        sleep(1);
    }

    //shutdown
    rclcpp::shutdown();
}